### PR TITLE
feat: [STO-4447]: Add STO Module to license registration

### PR DESCRIPTION
**Pull request by ahokinson**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33596)

### CODE REVIEW

File: 960-ng-license-beans/src/main/java/io/harness/serializer/kryo/LicenseBeanKryoClassesRegistrar.java [hunk 1]
```diff
@@ -28,6 +29,7 @@ public void register(Kryo kryo) {
     kryo.register(CFModuleLicenseDTO.class, 930003);
     kryo.register(CIModuleLicenseDTO.class, 930004);
     kryo.register(CVModuleLicenseDTO.class, 930005);
+    kryo.register(STOModuleLicenseDTO.class, 930010);
```
Comment (ahokinson) [inline 960-ng-license-beans/src/main/java/io/harness/serializer/kryo/LicenseBeanKryoClassesRegistrar.java:32]: I am unsure of this integer. Documentation online or within the project doesn't really explain this. Builds pass and checks pass. Is that enough?

File: 960-ng-license-beans/src/main/java/io/harness/licensing/beans/summary/STOLicenseSummaryDTO.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+@OwnedBy(HarnessTeam.STO)
+@Data
+@SuperBuilder
+@NoArgsConstructor
+@EqualsAndHashCode(callSuper = true)
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(NON_NULL)
+@Schema(name = "CILicenseSummary",
```
Comment (r0610205) [inline 960-ng-license-beans/src/main/java/io/harness/licensing/beans/summary/STOLicenseSummaryDTO.java:33]: Should be STO

File: 960-ng-license-beans/src/main/java/io/harness/licensing/beans/summary/STOLicenseSummaryDTO.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    description = "This contains details of the Continuous Integration License Summary in Harness")
```
Comment (r0610205) [inline 960-ng-license-beans/src/main/java/io/harness/licensing/beans/summary/STOLicenseSummaryDTO.java:34]: Continious Integration -> Security Testing Orchestration

File: 930-ng-license-manager/src/main/java/io/harness/licensing/helpers/ModuleLicenseSummaryHelper.java [hunk 4]
```diff
@@ -103,6 +105,19 @@ public static LicensesWithSummaryDTO generateSummary(
           }
         };
         break;
+      case STO:
+        licensesWithSummaryDTO = STOLicenseSummaryDTO.builder().build();
+        summaryHandler = (moduleLicenseDTO, summaryDTO, current) -> {
+          STOModuleLicenseDTO temp = (STOModuleLicenseDTO) moduleLicenseDTO;
+          STOLicenseSummaryDTO ciLicenseSummaryDTO = (STOLicenseSummaryDTO) summaryDTO;
```
Comment (r0610205) [inline 930-ng-license-manager/src/main/java/io/harness/licensing/helpers/ModuleLicenseSummaryHelper.java:112]: should be stoLicenseSummaryDTO

File: 930-ng-license-manager/src/main/java/io/harness/licensing/entities/modules/STOModuleLicense.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+@OwnedBy(HarnessTeam.STO)
+@Data
+@Builder
+@EqualsAndHashCode(callSuper = true)
+@Entity(value = "moduleLicenses", noClassnameStored = true)
+@Persistent
+@TypeAlias("io.harness.license.entities.module.STOModuleLicense")
+public class STOModuleLicense extends ModuleLicense {
+  private Integer numberOfCommitters;
```
Comment (r0610205) [inline 930-ng-license-manager/src/main/java/io/harness/licensing/entities/modules/STOModuleLicense.java:28]: That needs to be numberOfDevelopers

**Comment by r0610205** [comment]
*[`960-ng-license-beans/src/main/java/io/harness/serializer/kryo/LicenseBeanKryoClassesRegistrar.java` line 32 at r1](https://reviewable.io/reviews/harness/harness-core/33596#-N30Ak99ksSS1UbYEcL6-r1-32:-N31B_bq0EfH7qs5TZ9S:b-r9t08) ([raw file](https://github.com/harness/harness-core/blob/1d0ac09790717dabb49162eab703006f09ed8581/960-ng-license-beans/src/main/java/io/harness/serializer/kryo/LicenseBeanKryoClassesRegistrar.java#L32)):*
<details><summary><i>Previously, ahokinson (Anders Hokinson) wrote…</i></summary><blockquote>

I am unsure of this integer. Documentation online or within the project doesn't really explain this. Builds pass and checks pass. Is that enough?
</blockquote></details>

That's fine

**Review by r0610205 — APPROVED** [review]
> Reviewed 1 of 12 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33596)* status: :shipit: complete! ✓

**Review by r0610205 — DISMISSED** [review]
> Reviewed 1 of 12 files at r1, 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33596)* status: :shipit: complete! ✓

**Review by johannesHarness — APPROVED** [review]
> Reviewed 1 of 5 files at r4.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33596)* status: 2 file(s) to review.

**Review by williamwissemann — APPROVED** [review]
> Reviewed 2 of 12 files at r1, 2 of 2 files at r2, 6 of 7 files at r3, 4 of 5 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33596)* status: :shipit: complete! ✓

**Pull request merged**